File: backend/app/schemas/document.py
"""Pydantic schemas for document resources."""

from datetime import datetime

from pydantic import BaseModel


class DocumentOut(BaseModel):
    id: str
    original_name: str
    storage_path: str
    checksum: str
    size: int
    status: str
    expires_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True,
    }


class DocumentUploadResponse(BaseModel):
    document: DocumentOut
    is_duplicate: bool = False

File: frontend/src/types/document.ts
export type AnalysisStatus = 'not_started' | 'recognizing' | 'analyzing' | 'completed';

export interface Document {
  id: string;
  original_name: string;
  storage_path: string;
  checksum: string;
  size: number;
  status: string;
  expires_at: string | null;
  created_at: string;
  updated_at: string;
  // 分析状态相关字段
  analysis_status?: AnalysisStatus;
  analysis_progress?: number;
  session_id?: string;
}

export interface DocumentUploadResponse {
  document: Document;
  is_duplicate: boolean;
}
